tensorbay/client/job.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+
+"""Basic structures of asynchronous jobs."""
+
+from typing import Any, Callable, Dict, Optional, Tuple, Type, TypeVar
+
+from tensorbay.utility import AttrsMixin, ReprMixin, ReprType, attr, camel, common_loads
+
+_JOB_UPDATE_INTERVAL = 5
+
+
+class Job(AttrsMixin, ReprMixin):  # pylint: disable=too-many-instance-attributes
+    """This class defines :class:`Job`.
+
+    Arguments:
+        update_info: The function to get the latest job information.
+        title: Title of the Job.
+        job_id: ID of the Job.
+        job_type: Type of the Job.
+        arguments: Arguments of the Job.
+        created_at: The time when the Job is created.
+        started_at: The time when the Job is started.
+        finished_at: The time when the Job is finished.
+        status: The status of the Job.
+        error_message: The error message of the Job.
+        result: The result of the Job.
+        description: The description of the Job.
+
+    """
+
+    _T = TypeVar("_T", bound="Job")
+
+    _repr_type = ReprType.INSTANCE
+    _repr_maxlevel = 2
+
+    _repr_attrs: Tuple[str, ...] = (
+        "title",
+        "description",
```

Comment:
we do not put description in `_repr_attrs` due to it may be too long and has multiple lines.